### ISSUE TITLE
Power Query Activity - runtime issue with M query executed on spark engine at runtime when using a csv file that has #(cr) or #(lf) in a field.

### ISSUE THREAD

**Issue by JohnMiner3** [Azure/Azure-DataFactory#348]
Dear ADF Team,

There is an issue with the new Power Query Activity.  It does not handling CSV files that have quoted text fields that contain #(lf) or #(cr).  The GUI for Power Query works exactly as expected and produces the correct results.  This works fine with regular ADF pipelines but not at run time when the Power Query spark engine is engaged.

Business case for tech article

  Read super bowl data from Wikipedia and save to csv file in bronze quality zone.  Use Power Query to clean csv from bronze quality zone and store in silver quality zone.  Load the destination Azure SQL table with data from csv file in silver quality zone.

  There are four ways in which I tried solving the problem.  The third one breaks with the issue.

1 - Use the Web Table activity to read from Wikipedia and write to Azure SQL table.  (one copy activity)

2 - Use Web table activity to read read Wikipedia and write to CSV file.  Read from CSV file and write to Azure SQL table.  (two copy activities)

3 - Use Web table activity to read read Wikipedia and write to CSV file in the bronze.   Clear silver directory using a delete activity.  Use Power Query to read the bronze csv file and write a silver csv file.  Read from CSV file and write to Azure SQL table.  (two copy activities, one delete activity and one PQ activity).

This case looks like it works in the GUI but fails at runtime with a null value assertion.  The issue is that the spark engine does not parse the CSV file like normal ADF pipelines.  This is the reported bug.

4 - Use Web table activity to read read Wikipedia and write to Parquet file in the bronze.   Clear silver directory using a delete activity.  Use Power Query to read the bronze parquet file and write a silver parquet file.  Read from CSV file and write to Azure SQL table.  (two copy activities, one delete activity and one PQ activity).

This case works since parquet files are strongly typed.

In short, I have screen shots and the adf template that can be used to repo this issue.

Sincerely

John Miner
Data Platform MVP
john.miner@insight.com
